src/lib/coda_base/transaction_payload.ml

```diff
@@ -82,8 +98,11 @@ let gen =
   let%map receiver = Public_key.Compressed.gen
   and amount = Amount.gen
   and fee = Fee.gen
-  and nonce = Account_nonce.gen in
-  {receiver; amount; fee; nonce}
+  and nonce = Account_nonce.gen
+  and memo =
+    String.gen_with_length (Sha256.Digest.length_in_bits / 8) Char.gen
```

Comment:
the other gen you did `String.gen` and then `Sha256.Digest.digest`, that's probably better if we're forcing this to be a sha256 digest